contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -112,29 +116,82 @@ const (
 // AdmissionHookClientConfig contains the information to make a TLS
 // connection with the webhook
 type AdmissionHookClientConfig struct {
-	// Address of the external admission hook, could be a host string, 
-    // a host:port pair, or a URL.
-	Address string
+    // Service is a reference to the service for this webhook. It must communicate
+	// on port 443
+	Service ServiceReference
 	// CABundle is a PEM encoded CA bundle which will be used to validate webhook's server certificate.
 	CABundle []byte
 }
 
+// ServiceReference holds a reference to Service.legacy.k8s.io
+type ServiceReference struct {
+	// Namespace is the namespace of the service
+	Namespace string
+	// Name is the name of the service
+	Name string
+}
 ```
 
-## Synchronization of AdmissionControlConfiguration (**optional for alpha implement**)
+## Synchronization of AdmissionControlConfiguration 
 
 If the `initializer admission controller` and the `generic webhook admission
 controller` watch the `AdmissionControlConfiguration` and act upon deltas, their
 cached version of the configuration might be arbitrarily delayed. This makes it
 impossible to predict what initializer/hooks will be applied to newly created
 objects.
 
-We considered a few ways to make the behavior of the `initializer admission
+We propose two ways to make the behavior of the `initializer admission
 controller` and the `generic webhook admission controller` predictable.
 
-(I prefer #2. #1 is inefficient, #3 requires complex schema and is not intuitive)
+#### 1. Do consistent read of AdmissionControlConfiguration periodically
+
+The `initializer admission controller` and the `generic webhook admission
+controller` do a consistent read of the AdmissionControlConfiguration either 30s
+after the last read, or when there is a request that needs the two controllers to
+apply the configuration, whichever comes later.
+
+If the read fails, the two admission controllers block all incoming request.
+
+#### 2. Always do a consistent read of a smaller object
 
-#### 1. Always do consistent read 
+A consistent read of the AdmissionControlConfiguration object is expensive, we
+cannot do it for every incoming request.
+
+Alternatively, we record the resource version of the AdmissionControlConfiguration
+in a configmap. The apiserver that handles an update of the AdmissionControlConfiguration
+updates the configmap with the updated resource version. In the HA setup, there
+are multiple apiservers that update this configmap, they should only
+update if the recorded resource version is lower than the local one.
+
+The `initializer admission controller` and the `generic webhook admission
+controller` do a consistent read of the configmap *everytime* before applying
+the configuration to an incoming request. If the configmap has changed, then
+they do a consistent read of the `AdmissionControlConfiguration`.
+
+## What if an initializer controller/webhook is not ready after registered? (**optional for alpha implement**)
```

Comment:
this is unclear to me-- is the first step going to create things fail-open or fail-closed?

(I think we should do fail-open for 1.7 and add fail-closed behavior later when this feature grows up)